lib/matplotlib/category.py

```diff
@@ -1,123 +1,162 @@
-# -*- coding: utf-8 OA-*-za
+# -*- coding: utf-8 -*-
 """
 catch all for categorical functions
 """
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
+
+from collections import OrderedDict
+import itertools
+
 import six
 
+
 import numpy as np
 
 import matplotlib.units as units
 import matplotlib.ticker as ticker
 
 # np 1.6/1.7 support
 from distutils.version import LooseVersion
-import collections
-
-
-if LooseVersion(np.__version__) >= LooseVersion('1.8.0'):
-    def shim_array(data):
-        return np.array(data, dtype=np.unicode)
-else:
-    def shim_array(data):
-        if (isinstance(data, six.string_types) or
-                not isinstance(data, collections.Iterable)):
-            data = [data]
-        try:
-            data = [str(d) for d in data]
-        except UnicodeEncodeError:
-            # this yields gibberish but unicode text doesn't
-            # render under numpy1.6 anyway
-            data = [d.encode('utf-8', 'ignore').decode('utf-8')
-                    for d in data]
-        return np.array(data, dtype=np.unicode)
+
+VALID_TYPES = tuple(set(six.string_types +
+                        (bytes, six.text_type, np.str_, np.bytes_)))
 
 
 class StrCategoryConverter(units.ConversionInterface):
     @staticmethod
     def convert(value, unit, axis):
-        """Uses axis.unit_data map to encode
-        data as floats
+        """Uses axis.units to encode string data as floats
```

Comment:
This doc string isn't correct in this implementation.  It uses ``unit`` to encode the string.